diffrax/_term.py

```diff
@@ -776,3 +785,180 @@ def _to_vjp(_y, _diff_args, _diff_term):
         dy, vjp = jax.vjp(_to_vjp, y, diff_args, diff_term)
         da_y, da_diff_args, da_diff_term = vjp((-(a_y**ω)).ω)
         return dy, da_y, da_diff_args, da_diff_term
+
+
+# The Underdamped Langevin SDE trajectory consists of two components: the position
+# `x` and the velocity `v`. Both of these have the same shape.
+# So, by UnderdampedLangevinX we denote the shape of the x component, and by
+# UnderdampedLangevinTuple we denote the shape of the tuple (x, v).
+UnderdampedLangevinLeaf = Shaped[Array, "*underdamped_langevin"]
+UnderdampedLangevinX = PyTree[
+    Shaped[Array, "?*underdamped_langevin"], "UnderdampedLangevinX"
+]
+UnderdampedLangevinTuple = tuple[UnderdampedLangevinX, UnderdampedLangevinX]
+
+
+def _broadcast_pytree(source, target_tree):
```

Comment:
This function has appeared in two different spots now -- let's factor it out into `_misc.py`.